extensions/duplicate_message_policing.py

```diff
@@ -0,0 +1,75 @@
+"""
+Checks messages such that they are not duplicates of messages sent previously in the discord channel(s).
+This is designed to increase the effort required by users. For example, the string 'lol' can only be used
+once, and subsequent uses must get more creative (e.g. 'l0l', 'lolz', and so on).
+Eventually, some time in the future, all possible strings will have been said, and the database will need to be reset.
+
+Inspired by: https://blog.xkcd.com/2008/01/14/robot9000-and-xkcd-signal-attacking-noise-in-chat/
+"""
+import os, re
+import hikari, lightbulb
+import db
+import hashlib
+import sqlite3
+import re
+
+
+plugin = lightbulb.Plugin("duplicate_message_policing")
+
+
+@plugin.listener(hikari.GuildMessageCreateEvent)
+async def delete_duplicate(event: hikari.GuildMessageCreateEvent) -> None:
+    """
+    Deletes duplicate messages (excepting some). A duplicate message is simply a matching string
+    that has been seen before in the server (and not deleted).
+    """
+
+    nodelete_flag = "!"
+    # random rules. Probably worth thinking about this some more, if this bot function doesn't get deleted.
+    if (
+        (
+            event.channel_id != 813266892818219038 and os.environ.get("DEBUG") is None
+        )  # channel id is for #offtopic
+        or event.content is None
+        or event.is_webhook
+        or event.content.startswith(nodelete_flag)
+        or event.is_bot
+        or not event.content
+        or "http" in event.content  # allow links
+        or "@" in event.content  # allow mentions
+        or len(event.content) <= 4  # allow short messages
```

Comment:
This still allows the string `lol` :(

Reply:
A little more consideration makes me think the appropriate check might be `<= 2`, to allow emoji and most common ascii emoticons, while still prohibiting the dreaded `lol`.